cmd/contour-operator.go

```diff
@@ -61,6 +64,15 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
 
+	images := []string{contourImage, envoyImage}
+	for _, image := range images {
+		_, err := reference.Parse(image)
```

Comment:
Consider wrapping `reference.Parse` and adding a test to defend against the fragility that @Miciah pointed out.